File: Programming_Fundamentals_May_2016/Exercise/4.2. Lists And Matrices/DiagonalDifference.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace _03.Diagonal_Difference
{
    class DiagonalDifference
    {
        static void Main(string[] args)
        {
            int size = int.Parse(Console.ReadLine());

            int[][] matrix = new int[size][];

            FillMatrix(matrix, size);

            int primaryDiagonal = 0;
            int secondaryDiagonal = 0;

            for (int i = 0; i < size; i++)
            {
                primaryDiagonal += matrix[i][i];
                secondaryDiagonal += matrix[i][size - 1 - i];
            }

            int diff = Math.Abs(primaryDiagonal - secondaryDiagonal);
            Console.WriteLine(diff);
        }        

        private static void FillMatrix(int[][] matrix, int rows)
        {
            for (int row = 0; row < rows; row++)
            {
                matrix[row] = Console.ReadLine()
                    .Split()
                    .Select(int.Parse)
                    .ToArray();
            }
        }
    }
}

File: Java_Fundamentals_January_2017/Java_Advanced/Exersices/02. Matrices/src/DiagonalDifference.java
import java.util.Scanner;

public class DiagonalDifference {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        int size = Integer.parseInt(scan.nextLine());
        int[][] matrix = PrimaryDiagonal(scan, size);

        long primaryDiagonal = CalcPrimaryDiagonal(matrix, size);
        long secondaryDiagonal = CalcSecondaryDiagonal(matrix, size);

        long diff = Math.abs(primaryDiagonal - secondaryDiagonal);
        System.out.println(diff);
    }

    private static long CalcSecondaryDiagonal(int[][] matrix, int size) {
        long sum = 0;

        for (int i = 0; i < size; i++) {
            for (int j = size; j >= 0; j--) {
                if (j == size - i - 1){
                    sum += matrix[i][j];
                }
            }
        }

        return sum;
    }

    private static long CalcPrimaryDiagonal(int[][] matrix, int size) {
        long sum = 0;

        for (int i = 0; i < matrix.length; i++) {
            for (int j = 0; j < matrix[i].length; j++) {
                if (i == j){
                    sum += matrix[i][j];
                }
            }
        }

        return sum;
    }

    private static int[][] PrimaryDiagonal(Scanner scan, int size) {
        int[][]matrix = new int[size][size];

        for (int i = 0; i < size; i++) {
            String[] input = scan.nextLine().split(" ");
            for (int j = 0; j < size; j++) {
                matrix[i][j] = Integer.parseInt(input[j]);
            }
        }
        return matrix;
    }
}
